File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    int l = 0, w = 1, s = 0; // letter, word and sentence count
    string txt = get_string("Text: ");
    int txtlen = strlen(txt);

    for (int i = 0; i < txtlen; i++) // loop to count letters
    {
        if (isalpha(txt[i])) // is alphabet
        {
            l++;
        }
    }
    for (int i = 0; i < txtlen; i++) // loop to count words
    {
        if (isspace(txt[i]))
        {
            w++;
        }
    }
    for (int i = 0; i < txtlen; i++) // loop to count sentences
    {
        if (txt[i] == '.' || txt[i] == '?' || txt[i] == '!') // check for end sentence characters
        {
            s++;
        }
    }

    float calc = (0.0588 * l / w * 100) - (0.296 * s / w * 100) - 15.8; // readability formula calculation
    int idx = round(calc);

    if (idx < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (idx > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", idx);
    }
}

File: sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

# letters, words, sentences, counter
l = w = s = i = 0
length = len(text)

while i < length:
    # check if it's alphabet to add letters
    if text[i].isalpha():
        l += 1
    # check for spaces to add words
    if i == 0 and text[i] != " " or i != length and text[i] == " " and text[i+1] != " ":
        w += 1
    # check for periods, question or exclamation marks to add sentences
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        s += 1
    i += 1

L = l * 100 / w
S = s * 100 / w
idx = round(0.0588 * L - 0.296 * S - 15.8)

if idx < 1:
    print("Before Grade 1")
elif idx >= 16:
    print("Grade 16+")
else:
    print(f"Grade {idx}")